### ISSUE TITLE
Sending email asynchronously

### ISSUE THREAD

**Issue by rupalkhilari** [electology/approval_frame#63]
Noticed that django by default (django.core.email) waits for the email to be sent before navigating to the 'email sent' confirmation page. As a result, the application blocks for 2-3 minutes until the email is sent. Perhaps we should look into queuing the mail request or using another thread so that the user is not forced to wait on a specific page (registration, password reset) while the email is being sent. [This post] on stackoverflow (http://stackoverflow.com/questions/11957124/django-registration-sending-email-in-a-asynchronous-way) outlines a couple of options.

Thanks!

**Comment by ashimaathri** [comment]
I think that will be good to have. Some other posts I saw on this topic -

1. http://www.nomadblue.com/blog/django/async-mailing-django-amazon-ses-celery-supervisord/
2. http://www.dangtrinh.com/2013/04/asynchronously-sending-email-using.html
3. http://scanova.io/blog/engineering/2014/05/05/asynchronous-email-sending-using-django-post_office-celery/

Looks like there are many approaches to this problem! I'm not too familiar with any of them so you may have to read through all the material and make a decision on what to use. We should also ask Aaron about the infrastructure that electology uses for sending out email and if we could use that too.

**Comment by rupalkhilari** [comment]
@ashimaathri , yes thats a good idea. We should check to see what electology already uses. 
Thanks for the resources! I too haven't tried any of them yet so we'll have to evaluate.

**Comment by balajisubr** [comment]
@rupalkhilari I think Celery(http://docs.celeryproject.org/en/master/getting-started/first-steps-with-celery.html#first-steps) is a good option for this. Let me know your thoughts.

**Comment by rupalkhilari** [comment]
@balajisubr that sounds like a great option, and seems to be widely adopted for this task as well http://docs.celeryproject.org/en/latest/django/first-steps-with-django.html . Please go ahead and attempt this. 

Thanks!
